pkg/crc/cache/cache.go

```diff
@@ -80,12 +82,17 @@ func (c *Cache) CacheBinary() error {
 	if err != nil {
 		return err
 	}
-
-	// Extract the tarball and put it the cache directory.
-	extractedFiles, err := extract.UncompressWithFilter(assetTmpFile, tmpDir,
-		func(filename string) bool { return filepath.Base(filename) == c.binaryName })
-	if err != nil {
-		return errors.Wrapf(err, "Cannot uncompress '%s'", assetTmpFile)
+	var extractedFiles []string
+	// Check the file is tarball or not
+	if IsTarball(assetTmpFile) {
+		// Extract the tarball and put it the cache directory.
+		extractedFiles, err = extract.UncompressWithFilter(assetTmpFile, tmpDir,
+			func(filename string) bool { return filepath.Base(filename) == c.binaryName })
+		if err != nil {
+			return errors.Wrapf(err, "Cannot uncompress '%s'", assetTmpFile)
+		}
+	} else {
+		extractedFiles = append(extractedFiles, assetTmpFile)
```

Comment:
I think this should at least warn if `assetTmpFile` does not match `c.binaryName`?

Reply:
@cfergeau Added the warn message.